### PR TITLE
 Merge ImportKind and ExportKind enums into ExternalKind

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: lib/fizzy/types.hpp [hunk 1]
```diff
@@ -190,7 +190,7 @@ struct Import
     } desc;
 };
 
-enum class ExportType : uint8_t
+enum class ExportKind : uint8_t
```
Comment (axic) [inline lib/fizzy/types.hpp]: This is actually called export type in the spec, isn't it? But I think kind is a better term.

Comment (gumb0) [reply lib/fizzy/types.hpp]: There's no name for it in Binary Format, but in Structure there's a notion of "External type" (the actual type of function/global etc.) https://webassembly.github.io/spec/core/syntax/types.html#external-types
which suggests maybe we should merge `ImportKind` and `ExportKind` into `ExternalKind`

Comment (axic) [reply lib/fizzy/types.hpp]: Yes, sounds good.

File: lib/fizzy/types.hpp [hunk 2]
```diff
@@ -203,7 +203,7 @@ enum class ExportType : uint8_t
 struct Export
 {
     std::string name;
-    ExportType type = ExportType::Function;
+    ExportKind type = ExportKind::Function;
```
Comment (axic) [inline lib/fizzy/types.hpp]: Shouldn't this also use `kind`? `struct Import` names it kind.

**Pull request merged**